react/features/invite/functions.js

```diff
@@ -421,29 +422,90 @@ export function getShareInfoText(
 
     if (includeDialInfo) {
         const { room } = parseURIString(inviteUrl);
-        let dialInfoPageUrl = getDialInfoPageURL(room);
+        let numbersPromise;
+
+        if (state['features/invite'].numbers
+            && state['features/invite'].conferenceID) {
+            numbersPromise = Promise.resolve(state['features/invite']);
+        } else {
+            // we are requesting numbers and conferenceId directly
+            // not using updateDialInNumbers, because custom room
+            // is specified and we do not want to store the data
+            // in the state
+            const { dialInConfCodeUrl, dialInNumbersUrl, hosts }
+                = state['features/base/config'];
+            const mucURL = hosts && hosts.muc;
+
+            if (!dialInConfCodeUrl || !dialInNumbersUrl || !mucURL) {
+                // URLs for fetching dial in numbers not defined
+                return Promise.reject();
+            }
+
+            numbersPromise = Promise.all([
+                getDialInNumbers(dialInNumbersUrl),
+                getDialInConferenceID(dialInConfCodeUrl, room, mucURL)
+            ]).then(([ { defaultCountry, numbers }, {
+                conference, id, message } ]) => {
 
-        if (useHtml) {
-            dialInfoPageUrl
-                = `<a href="${dialInfoPageUrl}">${dialInfoPageUrl}</a>`;
+                if (!conference || !id) {
+                    return Promise.reject(message);
+                }
+
+                return {
+                    defaultCountry,
+                    numbers,
+                    conferenceID: id
+                };
+            });
         }
 
-        infoText += i18next.t('share.dialInfoText', { dialInfoPageUrl });
+        return numbersPromise.then(
+            ({ conferenceID, defaultCountry, numbers }) => {
+                const phoneNumber
+                    = _getDefaultPhoneNumber(numbers, defaultCountry) || '';
+
+                return `${
+                    i18next.t('info.dialInNumber')} ${
+                    phoneNumber} ${
+                    i18next.t('info.dialInConferenceID')} ${
+                    conferenceID}#\n\n`;
+            })
+            .catch(error =>
+                logger.error('Error fetching numbers or conferenceID', error))
+            .then(defaultDialInNumber => {
```

Comment:
if defaultDialInNumber is undefined due to an error, 'share.dialInfoText' will properly ignore it?

Reply:
That is correct.